sippy-ng/src/prow_job_runs/ProwJobRun.js

```diff
@@ -0,0 +1,926 @@
+  const [eventIntervals, setEventIntervals] = React.useState([])
+  const [filteredIntervals, setFilteredIntervals] = React.useState([])
+
+  // TODO: Not sure this needs to be a useState param, really just need a constant map but
+  // wasn't working.
+  const [allCategories, setAllCategories] = useState(
+    new Map([
+      ['operator_unavailable', 'Operator Unavailable'],
+      ['operator_progressing', 'Operator Progressing'],
+      ['operator_degraded', 'Operator Degraded'],
+      ['pods', 'Pods (careful)'],
+      ['pod_logs', 'Pod Logs'],
+      ['interesting_events', 'Interesting Events'],
+      ['alerts', 'Alerts'],
+      ['node_state', 'Node State'],
+      ['e2e_test_failed', 'E2E Failed'],
+      ['e2e_test_flaked', 'E2E Flaked'],
+      ['e2e_test_passed', 'E2E Passed'],
+      ['disruption', 'Disruption'],
+    ])
+  )
+
+  const [categoryButtonState, setCategoryButtonState] = useState(() => {
+    const params = new URLSearchParams(window.location.search)
+
+    let categories = []
+    let categoriesParam = params.get('categories')
+    if (categoriesParam) {
+      categories = categoriesParam.split(',')
+    }
+
+    const initialState = {}
+    let atLeastOneCategorySelected = false
+    allCategories.forEach((value, key, map) => {
+      initialState[key] = categories.includes(key)
+      if (categories.includes(key)) {
+        atLeastOneCategorySelected = true
+      }
+    })
+
+    if (!atLeastOneCategorySelected) {
+      console.log('no category was selected, enabling the default set')
+      initialState['operator_unavailable'] = true
+      initialState['operator_progressing'] = true
+      initialState['operator_degraded'] = true
+      initialState['pod_logs'] = true
+      initialState['interesting_events'] = true
+      initialState['alerts'] = true
+      initialState['node_state'] = true
+      initialState['e2e_test_failed'] = true
+      initialState['disruption'] = true
+    }
+
+    // If no buttons were selected, you're not going to see anything, and this likely implies
+    // you are fresh loading the page. Select the default set.
+    console.log('calculated initial category button state:')
+    console.log(initialState)
+    return initialState
+  })
+
+  const [intervalFiles, setIntervalFiles] = useState([])
+  const [selectedIntervalFiles, setSelectedIntervalFiles] = useState(() => {
+    const params = new URLSearchParams(window.location.search)
+    let tempFiles = []
+    let p = params.get('intervalFiles')
+    if (p) {
+      tempFiles = p.split(',')
+    }
+    return tempFiles
+  })
+
+  const [filterText, setFilterText] = useState(() => {
+    const params = new URLSearchParams(window.location.search)
+    if (params.get('filter')) {
+      return params.get('filter')
+    }
+    return ''
+  })
+
+  const fetchData = () => {
+    if (isLoaded) {
+      console.log(
+        'fetchData called due to UI update but intervals already loaded, re-using intervals'
+      )
+      return
+    }
```

Comment:
You could avoid this but having multiple `useEffect` hooks, so you could have one with no conditions so fetchdata gets called only one time, and then one that calls updatefiltering when  [categoryButtonState, history, selectedIntervalFiles]) gets  updated